neqo-http3/src/connection.rs

```diff
@@ -613,23 +670,83 @@ impl<E: Http3Events + Default, T: Http3Transaction, H: Http3Handler<E, T>>
         Ok(())
     }
 
-    fn handle_settings(&mut self, s: &[(HSettingType, u64)]) -> Res<()> {
-        qinfo!([self], "Handle SETTINGS frame.");
+    fn set_qpack_settings(&mut self, s: &[(HSettingType, u64)]) -> Res<()> {
         for (t, v) in s {
             qinfo!([self], " {:?} = {:?}", t, v);
             match t {
-                HSettingType::MaxHeaderListSize => {
-                    self.max_header_list_size = *v;
-                }
                 HSettingType::MaxTableSize => self.qpack_encoder.set_max_capacity(*v)?,
                 HSettingType::BlockedStreams => self.qpack_encoder.set_max_blocked_streams(*v)?,
-
                 _ => {}
             }
         }
         Ok(())
     }
 
+    fn handle_settings(&mut self, s: &[(HSettingType, u64)]) -> Res<()> {
+        qinfo!([self], "Handle SETTINGS frame.");
+        match &self.settings_state {
+            Http3RemoteSettingsState::NotReceived => {
+                self.set_qpack_settings(s)?;
+                self.settings_state = Http3RemoteSettingsState::Received(s.to_vec());
+                Ok(())
+            }
+            Http3RemoteSettingsState::ZeroRtt(settings) => {
+                let mut differ = false;
+                for st in &[
+                    HSettingType::MaxHeaderListSize,
+                    HSettingType::MaxTableSize,
```

Comment:
This should be called MaxTableCapacity.  The spec uses SETTINGS_QPACK_MAX_TABLE_CAPACITY.  Note also that this is a value that [cannot change](https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#section-3.2.3) if the value is non-zero.